app/src/pages/PageProjects.re

```diff
@@ -15,63 +17,51 @@ module List = {
     let ps =
       Array.map(
         project =>
-          <Link page={Router.Project(project.address)}>
-            <ProjectCard
-              imgUrl={project.imgUrl}
-              name={project.name}
-              description={project.description}
-            />
-          </Link>,
+          <li key={project.address}>
+            <Link page={Router.Project(project.address)}>
+              <ProjectCard
+                imgUrl={project.imgUrl}
+                name={project.name}
+                description={project.description}
+              />
+            </Link>
+          </li>,
         projects,
       );
 
     <ul> {React.array(ps)} </ul>;
   };
 };
 
-type action =
-  | ProjectsFetched(array(project));
-
-type state =
-  | Loading
-  | Fetched(array(project))
-  | Failed(string);
-
 [@react.component]
 let make = () => {
-  let (state, dispatch) =
-    React.useReducer(
-      (_state, action) =>
-        switch (action) {
-        | ProjectsFetched(ps) => Fetched(ps)
-        },
-      Loading,
-    );
+  let state = Store.useSelector(state => state.projectsState);
+  let dispatch = Store.useDispatch();
 
-  React.useEffect0(() => {
-    getProjects()
-    |> Js.Promise.then_(projects =>
-         ProjectsFetched(projects) |> dispatch |> Js.Promise.resolve
-       )
-    |> ignore;
+  Js.log(state);
 
-    None;
-  });
+  if (state.projects == None) {
+    dispatch(StoreMiddleware.Thunk(ThunkProjects.fetchProjects)) |> ignore;
+  };
 
   <>
     <div className=Styles.projectHeading>
       <Container.TwoColumns>
         ...(
              <Title.Huge> {React.string("Explore")} </Title.Huge>,
-             <Button> {React.string("Register project")} </Button>,
+             <Link page=Router.RegisterProject>
+               <Button> {React.string("Register project")} </Button>
+             </Link>,
            )
       </Container.TwoColumns>
     </div>
     {
-      switch (state) {
-      | Loading => <div> {React.string("Loading...")} </div>
-      | Fetched(projects) => <List projects />
-      | Failed(_error) => <div className="error" />
+      switch (state.error, state.loading, state.projects) {
+      | (Some(error), _, _) =>
+        <div className="error"> {React.string("ERROR: " ++ error)} </div>
+      | (None, false, Some(projects)) => <List projects />
+      | (None, true, _) => <div> {React.string("Loading...")} </div>
+      | _ => <div> {React.string("Not loading...")} </div>
```

Comment:
Should the fall-through case perhaps also be an error or an exception?

Reply:
I believe it's more accurate to treat it as the initial or default state. Going forward we should push hard to have designs and flows represent these states so we get in the habit of accounting for them.

An error should only have occurred if the states error field is set.